sycl/include/sycl/detail/property_list_base.hpp

```diff
@@ -92,15 +92,16 @@ class PropertyListBase {
   get_property_helper() const {
     const int PropKind = static_cast<int>(PropT::getKind());
     if (PropKind >= PropWithDataKind::PropWithDataKindSize)
-      throw sycl::invalid_object_error("The property is not found",
-                                       PI_ERROR_INVALID_VALUE);
+      detail::throw_exception(errc::invalid, "The property is not found");
 
     for (const std::shared_ptr<PropertyWithDataBase> &Prop : MPropsWithData)
       if (Prop->isSame(PropKind))
         return *static_cast<PropT *>(Prop.get());
 
-    throw sycl::invalid_object_error("The property is not found",
-                                     PI_ERROR_INVALID_VALUE);
+    detail::throw_exception(errc::invalid, "The property is not found");
+    // The above will throw an exception, but compilers complain about no return
+    // value. So, create an unreachable dummy throw to make it silent.
+    throw "dummy";
```

Comment:
Nit; Maybe add a "TODO" to use [`std::unreachable`](https://en.cppreference.com/w/cpp/utility/unreachable) when we can?

Reply:
Done. Thanks.